packages/client/src/debounce.test.tsx

```diff
@@ -0,0 +1,56 @@
+import debounce from "./debounce"
+
+jest.useFakeTimers()
+
+describe("debounce", () => {
+  let func: jest.Mock
+  let debouncedFunc: Function
+
+  beforeEach(() => {
+    func = jest.fn()
+    debouncedFunc = debounce(func, 3000, true)
+  })
+
+  it("executes once over a period shorter than 3 seconds", () => {
+    debouncedFunc()
+    expect(func).toBeCalledTimes(1)
+
+    for (let i = 0; i < 300; i++) {
+      debouncedFunc()
+    }
+
+    jest.runAllTimers()
+
+    expect(func).toBeCalledTimes(1)
+  })
+
+  it("calls func every 3 seconds over a period longer than 3 seconds", () => {
+    debouncedFunc()
+    expect(func).toBeCalledTimes(1)
+
+    for (let i = 0; i < 10; i++) {
+      debouncedFunc()
+      jest.advanceTimersByTime(3000)
+    }
+
+    expect(func).toBeCalledTimes(10)
+  })
+
+  it("eventually calls func with the same argument as the last call to debouncedFunc", () => {
+    let i
+    let args: number[] = []
+
+    for (i = 0; i < 5; i++) {
+      const rando = Math.floor(Math.random() * 6 + 1)
+      debouncedFunc(rando)
+      args.push(rando)
+      jest.advanceTimersByTime(3000)
```

Comment:
This is close to what I want. But because this test advances timers after every call we don't test cases where a call is delayed to the next three-second window.

This test should call `debouncedFunc` multiple times before advancing timers, then verify that `func` was called with the last argument after advancing timers.